src/dotnet/ToolPackage/CommandNameAllReservedValidator.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.DotNet.ToolPackage
+{
+    public static class CommandNameAllReservedValidator
+    {
+        private static CommandNameValidator[] _explicitReservedStrings = new CommandNameValidator[] {
+            new CommandNameValidator(false, false, true, "add"),
```

Comment:
super-super-nit: could this possibly be made more readable by using either named parameters or a flag enum instead of a lot of true/false literals?

Reply:
the spec is like that :(

Reply:
> the spec is like that :(

I agree the proposal is somewhat complicated and I hope we simplify it.

But that's not relevant to @dasMulli's comment. You can keep the exact same spec and logic and use either a flags enum or named arguments. I also find unnamed bool arguments hard to read and 3 consecutive bools are especially easy to get confused.

Reply:
Maybe add param name can help. Or just add comment some that mimic the header of the spreadsheet.

The current form is very easy to create from the spreadsheet use sublime ctrl+shift+l and alt+3, so it is less likely to "just copy it wrong". Also, the shape is the same as the spreadsheet. It is easy to edit accordingly if the sheet is change.

Reply:
Add comment, is that better?